hls4ml/backends/catapult/passes/conv_stream.py

```diff
@@ -6,7 +6,12 @@ class GenerateConvStreamingInstructions(OptimizerPass):
     '''Generates the instructions for streaming implementation of CNNs'''
 
     def match(self, node):
-        return isinstance(node, (Conv1D, SeparableConv1D, Conv2D, SeparableConv2D))
+        is_match = (
+            isinstance(node, (Conv1D, SeparableConv1D, Conv2D, SeparableConv2D))
+            and node.model.config.get_config_value('IOType').casefold() == 'io_stream'
```

Comment:
We haven't used casefold in other places. People has suggested to use 'lower' for consistency with other parts of the code.